### PR TITLE
use http param instead of init parameter to control range-header behavior

### PR DESCRIPTION
**Pull request by trondhindenes**
as discussed in https://github.com/piccolo-orm/piccolo_api/pull/122

### CODE REVIEW

**Comment by sinisaos** [comment]
@trondhindenes @dantownsend You were both right. This is a much cleaner approach. Can you add a description of the `__range_header` and `__range_header_name` query params to the `_split_params` method docstring. Thanks.

**Comment by trondhindenes** [comment]
> Can you add a description of the `__range_header` and `__range_header_name` query params to the `_split_params` method docstring. Thanks.

Done!

File: piccolo_api/fastapi/endpoints.py [hunk 1]
```diff
@@ -506,6 +506,37 @@ def modify_signature(
...
+            parameters.extend(
+                [
+                    Parameter(
+                        name="__range_header",
+                        kind=Parameter.POSITIONAL_OR_KEYWORD,
+                        annotation=bool,
+                        default=Query(
+                            default=False,
+                            description=(
+                                "enable to add range-header respons header"
```
Comment (dantownsend) [inline piccolo_api/fastapi/endpoints.py:518]: For the description, how about:

`Set to 'true' to add the Content-Range response header.`

File: piccolo_api/fastapi/endpoints.py [hunk 2]
```diff
@@ -506,6 +506,37 @@ def modify_signature(
...
+            parameters.extend(
+                [
+                    Parameter(
+                        name="__range_header_name",
+                        kind=Parameter.POSITIONAL_OR_KEYWORD,
+                        annotation=str,
+                        default=Query(
+                            default=None,
+                            description=(
+                                "object name in range-header reponse header"
```
Comment (dantownsend) [inline piccolo_api/fastapi/endpoints.py:533]: For the description, how about:

`Specify the object name in the Content-Range response header (defaults to the table name).`

**Comment by dantownsend** [comment]
@trondhindenes This looks great, thanks!

I just left some minor suggestions for the Swagger docs descriptions.

**Comment by trondhindenes** [comment]
@dantownsend thanks! Your suggestions make total sense - added those now.

**Comment by dantownsend** [comment]
Cool, looks good - sorry for all the back and forth. I think this is the right solution.

**Comment by trondhindenes** [comment]
@dantownsend there's a bug in this release - if `__range_header` is false it gets passed down to the model, which causes a validation error. Sorry about this, I'll create a hotfix PR right away.

**Comment by trondhindenes** [comment]
https://github.com/piccolo-orm/piccolo_api/pull/125. Again, sorry for this I should have testet for that from the get-go.

**Comment by dantownsend** [comment]
@trondhindenes No worries, these things happen. I'll release the hot fix when the build passes.

**Pull request merged**